structuralcodes/materials/constitutive_laws.py

```diff
@@ -0,0 +1,165 @@
+"""Collection of some standard constitutive laws"""
+import typing as t
+import numpy as np
+from numpy.typing import ArrayLike
+from ..core.base import ConstitutiveLaw
+
+
+class Elastic(ConstitutiveLaw):
+    """Class for elastic Constitutive Law"""
+
+    __materials__: t.Tuple[str] = (
+        'concrete',
+        'steel',
+        'rebars',
+    )
+
+    def __init__(self, E: float, name: t.Optional[str] = None) -> None:
+        """Initialize an Elastic Material"""
+        name = name if name is not None else "ElasticLaw"
+        super().__init__(name=name)
+        self._E = E
+
+    def get_stress(self, eps: ArrayLike) -> float:
+        """Return stress given strain"""
+        eps = np.asarray(eps)
+        return self._E * eps
+
+    def get_tangent(self, eps: ArrayLike) -> float:
+        """Return the tangent"""
+        return self._E
+
+
+class ElasticPlastic(ConstitutiveLaw):
+    """Class for elastic-plastic Constitutive Law"""
+
+    __materials__: t.Tuple[str] = (
+        'steel',
+        'rebars',
+    )
+
+    def __init__(
+        self,
+        E: float,
+        fy: float,
+        Eh: t.Optional[float] = None,
+        eps_su: t.Optional[float] = None,
+        name: t.Optional[str] = None,
+    ) -> None:
+        """Initialize an Elastic-Plastic Material"""
+        name = name if name is not None else "ElasticPlasticLaw"
+        super().__init__(name=name)
+        self._E = E
+        self._fy = fy
+        if Eh is not None:
+            self._Eh = Eh
+        else:
+            self._Eh = 0.0
+        self._eps_su = eps_su
+        self._eps_sy = fy / E
```

Comment:
Consider catching the `ZeroDivisionError` in case `E` is accidentally set to zero.